composer/metrics/metrics.py

```diff
@@ -0,0 +1,185 @@
+            self.total_union += area_prediction + area_target - area_intersect
+
+    def compute(self):
+        """Aggregate state across all processes and compute final metric."""
+        return 100 * (self.total_intersect / self.total_union).mean()  #type: ignore (third-party)
+
+
+class Dice(Metric):
+    """The Dice Coefficient for evaluating image segmentation.
+
+    The Dice Coefficient measures how similar predictions and targets are.
+    More concretely, it is computed as 2 * the Area of Overlap divided by
+    the total number of pixels in both images.
+
+    Args:
+        num_classes (int): the number of classes in the segmentation task.
+    """
+
+    def __init__(self, num_classes: int):
+        super().__init__(dist_sync_on_step=True)
+        self.add_state("n_updates", default=torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state("dice", default=torch.zeros((num_classes,)), dist_reduce_fx="sum")
+
+    def update(self, preds: Tensor, targets: Tensor):
+        """Update the state based on new predictions and targets."""
+        self.n_updates += 1  # type: ignore
+        self.dice += self.compute_stats(preds, targets)
+
+    def compute(self):
+        """Aggregate the state over all processes to compute the metric."""
+        dice = 100 * self.dice / self.n_updates  # type: ignore
+        best_sum_dice = dice[:]
+        top_dice = round(torch.mean(best_sum_dice).item(), 2)
+        return top_dice
+
+    @staticmethod
+    def compute_stats(preds: Tensor, targets: Tensor):
+        num_classes = preds.shape[1]
+        scores = torch.zeros(num_classes - 1, device=preds.device, dtype=torch.float32)
+        for i in range(1, num_classes):
+            if (targets != i).all():
+                # no foreground class
+                _, _pred = torch.max(preds, 1)
+                scores[i - 1] += 1 if (_pred != i).all() else 0
+                continue
+            _tp, _fp, _tn, _fn, _ = _stat_scores(preds, targets, class_index=i)  # type: ignore
+            denom = (2 * _tp + _fp + _fn).to(torch.float)
+            score_cls = (2 * _tp).to(torch.float) / denom if torch.is_nonzero(denom) else 0.0
+            scores[i - 1] += score_cls
+        return scores
+
+
+def _stat_scores(
+    preds: Tensor,
+    targets: Tensor,
+    class_index: int,
+    argmax_dim: int = 1,
+) -> Tuple[Tensor, Tensor, Tensor, Tensor, Tensor]:
+
+    if preds.ndim == targets.ndim + 1:
+        preds = to_categorical(preds, argmax_dim=argmax_dim)
+
+    tp = ((preds == class_index) * (targets == class_index)).to(torch.long).sum()
+    fp = ((preds == class_index) * (targets != class_index)).to(torch.long).sum()
+    tn = ((preds != class_index) * (targets != class_index)).to(torch.long).sum()
+    fn = ((preds != class_index) * (targets == class_index)).to(torch.long).sum()
+    sup = (targets == class_index).to(torch.long).sum()
+
+    return tp, fp, tn, fn, sup
+
+
+class CrossEntropyLoss(Metric):
```

Comment:
Maybe it makes sense to refer to this as just `CrossEntropy` instead of `CrossEntropyLoss` if it is meant to only be used as a metric and not a loss during training. Otherwise I'd expect to find it `loss.py`.